src/Sylius/Bundle/CoreBundle/PaymentRequest/Provider/ActionsHttpResponseProvider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\PaymentRequest\Provider;
+
+use Sylius\Bundle\PaymentBundle\Exception\PaymentRequestNotSupportedException;
+use Sylius\Bundle\ResourceBundle\Controller\RequestConfiguration;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+final class ActionsHttpResponseProvider implements ServiceProviderAwareProviderInterface
+{
+    /**
+     * @param ServiceProviderInterface<HttpResponseProviderInterface> $locator
+     */
+    public function __construct(
+        private ServiceProviderInterface $locator,
+    ) {
+    }
+
+    public function getResponse(
+        RequestConfiguration $requestConfiguration,
+        PaymentRequestInterface $paymentRequest,
+    ): Response {
+        $action = $paymentRequest->getAction();
+        $httpResponseProvider = $this->getHttpResponseProvider($action);
+        if (null === $httpResponseProvider) {
+            throw new PaymentRequestNotSupportedException(sprintf(
+                'No payment request HTTP Response provider supported for this action "%s".',
+                $action
+            ));
+        }
+
+        return $httpResponseProvider->getResponse($requestConfiguration, $paymentRequest);
+    }
+
+    public function supports(
+        RequestConfiguration $requestConfiguration,
+        PaymentRequestInterface $paymentRequest,
+    ): bool {
+        $action = $paymentRequest->getAction();
+        $httpResponseProvider = $this->getHttpResponseProvider($action);
+        if (null === $httpResponseProvider) {
+            return false;
+        }
+
+        return $httpResponseProvider->supports($requestConfiguration, $paymentRequest);
+    }
+
+    public function getProviderIndex(): array {
+        return $this->locator->getProvidedServices();
+    }
```

Comment:
```suggestion
    public function getProviderIndex(): array
    {
        return $this->locator->getProvidedServices();
    }
```
It seems like this one and the method below are heavily duplicated ATM, maybe we should extract some abstract class for this logic and template it for static analysis?

Reply:
They are using the same behaviour used in the `CommandProvider` but they don't have the same return types.

Reply:
I refactored to two abstract classes the command providers and the http response providers. Let me know if you see any other improvement to do here @NoResponseMate.